app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,42 @@
+class ReviewsController < ApplicationController
+  before_action :authenticate_user!
+
+  def edit
+    @sitter = User.find(params[:sitter_id])
+    @review = @sitter.reviews.find(params[:id])
+  end
+
+  def create
+    @sitter = User.find(params[:sitter_id])
+    @review = Review.new(review_params)
+    @user_review = UserReview.new(review: @review, user: current_user)
+    @sitter_review = UserReview.new(review: @review, user: @sitter)
+
+    if @review.save && @user_review.save && @sitter_review.save
+      flash[:notice] = "Review added successfully."
+      redirect_to sitter_path(@sitter)
+    else
+      @errors = @review.errors.full_messages
+      render "sitters/show"
+    end
+  end
+
+  def update
+    @sitter = User.find(params[:sitter_id])
+    @review = Review.find(params[:id])
+
+    if @review.update(review_params)
+      flash[:notice] = "Review updated!"
+      redirect_to sitter_path(@sitter)
+    else
+      @errors = @review.errors.full_messages
```

Comment:
No need to create an assignment here in the controller. Just call `@review.errors.full_messages` in the view and display accordingly.